pyfolio/utils.py

```diff
@@ -156,12 +167,24 @@ def get_returns_cached(filepath, update_func, latest_dt, **kwargs):
     if update_cache:
         returns = update_func(**kwargs)
         try:
-            mkdir(cache_dir())
+            ensure_directory(cache_dir())
+        except OSError as e:
+            warnings.warn(
+                'could not update cache: {}. {}: {}'.format(
+                    filepath, type(e).__name__, e,
+                ),
+                UserWarning,
+            )
+
+        try:
             returns.to_csv(filepath)
-        except IOError as e:
-            warnings.warn('Could not update cache {}.'
-                          'Exception: {}'.format(filepath, e),
-                          UserWarning)
+        except Exception as e:
```

Comment:
Do we want to be catching things like `MemoryError` though?  In `zipline.data.loader` we do this:
```
    # Stuff involving pandas to_csv/from_csv
    except (OSError, IOError, ValueError) as e:
        # These can all be raised by various versions of pandas on various
        # classes of malformed input.  Treat them all as cache misses.
        logger.info(
            "Loading data for {path} failed with error [{error}].".format(
                path=path, error=e,
            )
        )
```